common/src/main/scala/com/gu/recipeasy/db/DB.scala

```diff
@@ -319,6 +319,32 @@ class DB(contextWrapper: ContextWrapper) {
     )
   }
 
+  def userEventsDates(): List[String] = {
+    contextWrapper.dbContext.run(
+      quote(
+        query[UserEventDB].schema(_.entity("user_events")).map(event => event.event_datetime)
+      )
+    ).map(datetime => {
+        val format = new java.text.SimpleDateFormat("yyyy-MM-dd")
+        format.format(datetime)
+      }).distinct
+  }
+
+  def eventsForDateAndOperationType(date: String, opType: OperationType): List[UserEventDB] = {
+    val format = new java.text.SimpleDateFormat("yyyy-MM-dd")
+    contextWrapper.dbContext.run(
+      quote(
+        query[UserEventDB].schema(_.entity("user_events"))
+      )
+    ).filter(event => (format.format(event.event_datetime) == date) && (event.operation_type == opType.toString()))
+  }
+
+  def dailyActivityDistribution(): List[(String, Int, Int, Int)] = {
```

Comment:
This is quite confusing a type `List[(String, Int, Int, Int)]` and would probably be better modelled as a case class.